sklearn/preprocessing/label.py

```diff
@@ -147,7 +131,7 @@ def transform(self, y):
         if len(np.intersect1d(classes, self.classes_)) < len(classes):
```

Comment:
This should be apparent from `KeyErrors` now. Handle it in an `except` case (and avoid a pass over the data in the normal case).